cli/src/utils/downloadBin.ts

```diff
@@ -21,82 +26,221 @@ async function latestRelease(owner: string, repo: string) {
+          throw new Error(`Architecture ${arch} is not supported.`);
+      }
+    case "linux":
+      switch (arch) {
+        case "x64":
+          return System.LinuxAmd64;
+        case "arm":
+        // fallthrough
+        case "arm64":
+          return System.LinuxArm64;
+        default:
+          throw new Error(`Architecture ${arch} is not supported.`);
+      }
   }
 
-  return `${platformName}-${archName}`;
+  throw new Error(`Platform ${platform} is not supported.`);
 }
 
 function makeExecutable(filePath: string): void {
   fs.chmodSync(filePath, "755");
 }
 
-export async function downloadFileIfNotExists({
-  filePath,
+/**
+ * Download a binary from the given release artifact of the GitHub repository,
+ * if it was not already downloaded.
+ * @param localFilePath - The path to the local file (which either already
+ * exists or will be created).
+ * @param dirPath - The path to the directory where the file will be created.
+ * @param owner - The owner of the GitHub repository.
+ * @param repoName - The name of the GitHub repository.
+ * @param remoteFileName - The name of the file in the GitHub release artifact.
+ * @returns {Promise<void>}
+ */
+export async function downloadBinIfNotExists({
+  localFilePath,
   dirPath,
+  owner,
   repoName,
-  fileName,
+  remoteFileName,
 }: {
-  filePath: string;
+  localFilePath: string;
   dirPath: string;
+  owner: string;
   repoName: string;
-  fileName: string;
+  remoteFileName: string;
 }) {
   if (!fs.existsSync(dirPath)) {
     fs.mkdirSync(dirPath, { recursive: true });
   }
 
   // Check if file exists
-  if (await fileExists(filePath)) {
+  if (await fileExists(localFilePath)) {
     return;
   }
 
-  const system = getSystem();
-  const tag = await latestRelease("lightprotocol", repoName);
-
-  const url = `https://github.com/Lightprotocol/${repoName}/releases/download/${tag}/${fileName}-${system}`;
-
-  if (!url) {
-    throw new Error(`No binary found for the detected system ${system}`);
-  }
+  const tag = await latestRelease(owner, repoName);
+  const url = `https://github.com/${owner}/${repoName}/releases/download/${tag}/${remoteFileName}`;
 
   // Download the file
-  console.log(`Downloading ${fileName} from ${url}...`);
-  const { data } = await axios({
+  console.log(`Downloading ${remoteFileName} from ${url}...`);
+  const { data, headers } = await axios({
     url,
     method: "GET",
     responseType: "stream",
   });
 
-  // Save the file
-  const writer = fs.createWriteStream(filePath);
-  data.pipe(writer);
+  const totalLength = headers["content-length"];
 
-  return new Promise<void>((resolve, reject) => {
-    writer.on("finish", () => {
-      makeExecutable(filePath); // Make the file executable after it has been written.
-      resolve();
+  let progressStream = progress({
+    length: totalLength,
+    time: 100 /* ms */,
+  });
+
+  progressStream.on("progress", (progress: any) => {
+    console.log(`Progress: ${progress.percentage.toFixed(2)}%`);
```

Comment:
I was trying to find some proper library for showing a progress bar (like, a proper one, not just percentage as text), didn't find anything that great yet.But I'll keep looking.

Reply:
@sleepyqadir do you know such a library by any chance?